File: editor/en/[452]Minimum Number of Arrows to Burst Balloons.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda a: a[1])
        result = 1
        reach = points[0][1]

        for point in points:
            if point[0] > reach:
                result += 1
                reach = point[1]

        return result

# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[452]Minimum Number of Arrows to Burst Balloons.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
        int result = 1;
        int reach = points[0][1];

        for (int[] point : points) {
            if (point[0] > reach) {
                result++;
                reach = point[1];
            }
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
